### PR TITLE
fix: DiffScene Page Displays Accurate Number of Diff Entries

### PR DESCRIPTION
**Pull request by patnir41**
When comparing two specs in APIExplorer's Diff Scene, the scene will display the correct number of diff entries according to the page size, which is 15 by default.

### CODE REVIEW

File: packages/api-explorer/src/scenes/DiffScene/DocDiff/DocDiff.spec.tsx [hunk 1]
```diff
@@ -109,4 +109,30 @@ describe('DocDiff', () => {
       ).toBeInTheDocument()
     })
   })
+
+  it('paginates with correct number of entries per page', async () => {
```
Comment (jkaster) [inline packages/api-explorer/src/scenes/DiffScene/DocDiff/DocDiff.spec.tsx:113]: is it worth checking the last row isn't the first row on the next page?

Comment (patnir41) [reply packages/api-explorer/src/scenes/DiffScene/DocDiff/DocDiff.spec.tsx:113]: Yes I agree and feel it wouldn't hurt. See the new additional test.

File: packages/api-explorer/src/scenes/DiffScene/DocDiff/DocDiff.spec.tsx [hunk 2]
```diff
@@ -109,4 +109,61 @@ describe('DocDiff', () => {
...
+        leftKey={leftKey}
+        leftSpec={leftApi}
+        rightKey={rightKey}
+        rightSpec={rightApi}
+        delta={delta}
+        pageSize={pageSize}
+      />
+    )
+
+    for (let i = 0; i < pageSize; i++) {
```
Comment (josephaxisa) [inline packages/api-explorer/src/scenes/DiffScene/DocDiff/DocDiff.spec.tsx:127]: nit: our TS guide suggests using `for..of` or `forEach` for looping over iterables.
```suggestion
    for (row of delta) {
```

**Review by josephaxisa — COMMENTED** [review]
thanks Niral. One minor suggestion below

File: packages/api-explorer/src/scenes/DiffScene/DocDiff/DocDiff.spec.tsx (hunk 2 above)
Comment (josephaxisa) [reply packages/api-explorer/src/scenes/DiffScene/DocDiff/DocDiff.spec.tsx:127]: actually, I see how this is necessary given the parameters.

**Review by josephaxisa — APPROVED** [review]
lgtm

**Review by jkaster — APPROVED** [review]
LGTM

**Pull request merged**